app/views/my_facilities/bp_controlled.csv.erb

```diff
@@ -0,0 +1,42 @@
+<% rate_name = "controlled_patients_rate" %>
+<% arr = [] %>
+<% header = ["Facilities", "Total assigned", "Total registered","6-month change"] %>
+<% (@start_period..@period).each do |period| %>
+<% header << period #MONTHLY PERCENTAGE HEADER%>
+<% header << "Ratio" #MONTHLY RATIO HEADER %>
+<%end%>
+<%arr << header%>
+<% @display_sizes.each_with_index do |size, i| #For loop for the top row/ Sites of that size %>
+  <% row = []%>
+  <% six_month_rate_change = facility_size_six_month_rate_change(@stats_by_size[size][:periods], rate_name) %>
+  <% row << "All #{Facility.localized_facility_size(size)}s" #FACILITIES%>
+  <% row << number_or_zero_with_delimiter(@stats_by_size[size][:periods][@period][:cumulative_assigned_patients]) #Assigned pt %>
+  <% row << number_or_zero_with_delimiter(@stats_by_size[size][:periods][@period][:cumulative_registrations]) #Registered pt %>
+  <% row <<  number_to_percentage(six_month_rate_change, precision: 0) #6-MONTH CHANGE %>
+  <% @stats_by_size[size][:periods].each_pair do |period, data| %>
+    <% controlled_patients_rate = data["controlled_patients_rate"] %>
+    <% row << number_to_percentage(controlled_patients_rate || 0, precision: 0) #Monthly rate change %>
+    <% row << "#{data["controlled_patients"]} / #{data["adjusted_patient_counts"]}" #Monthly ratio %>
+  <% end %>
+  <%arr << row%>
+  <% @data_for_facility.each do |_, facility_data| #For loop for subsequent rows%>
+    <% sub_row = []%>
+    <% facility = facility_data.region.source %>
+    <% next if facility.facility_size != size #Explain why this is necessary??? structure of DB %>
+    <% six_month_rate_change = six_month_rate_change(facility, "controlled_patients_rate") %>
+    <% sub_row << facility.name #Facility name %>
+    <% sub_row << number_or_zero_with_delimiter(facility_data["cumulative_assigned_patients"].values.last) #Assigned pt %>
+    <% sub_row << number_or_zero_with_delimiter(facility_data["cumulative_registrations"].values.last)#Registered pt %>
+    <% sub_row << number_to_percentage_with_symbol(six_month_rate_change, precision: 0) #6-month change %>
+    <% (@start_period..@period).each do |period| %>
+      <% controlled_patients_rate = facility_data["controlled_patients_rate"][period] %>
+      <% sub_row << number_to_percentage(controlled_patients_rate || 0, precision: 0) #Monthly rate %>
+      <% sub_row << "#{facility_data["controlled_patients"][period]} / #{facility_data["adjusted_patient_counts"][period]}" #Monthly ratio %>
+    <% end %>
+    <%arr << sub_row%>
+  <% end %>
+  <% arr << [] if i === 0 #Partition between the facility sizes, for clarity %>
+<%end%>
+<%arr.each do |row|%>
+  <%= row.join(",")%>
```

Comment:
Joining data like this isn't a good way to generate CSVs due to escaping and quoting issues.

You should take a look at Ruby's builtin CSV library, it has a decent [API for writing](https://ruby-doc.org/stdlib-2.6.3/libdoc/csv/rdoc/CSV.html#class-CSV-label-Writing) CSV data to a string or a file.